### ISSUE TITLE
[Deeplab v3+] Training model from scratch

### ISSUE THREAD

**Issue by am-gill** [tensorflow/models#5663]
I want to train deeplab v3+ from scratch, without even using imagenet pretrained checkpoint. With this setting, I am not being able to get the model to converge. I tried changing learning rate, learning rate policy, optimization with Adam, but nothing seems to help. I would appreciate any help with this.
Thanks!

**Comment by choigyumin** [comment]
can you show your full script for run?

**Comment by asgill** [comment]
> can you show your full script for run?

I'd use something like:

```
python "${WORK_DIR}"/train.py \
  --logtostderr \
  --train_split="trainval" \
  --model_variant="xception_65" \
  --atrous_rates=6 \
  --atrous_rates=12 \
  --atrous_rates=18 \
  --output_stride=16 \
  --decoder_output_stride=4 \
  --train_crop_size=513 \
  --train_crop_size=513 \
  --train_batch_size=4 \
  --training_number_of_steps=300000 \
  --fine_tune_batch_norm=true \
  --train_logdir="${TRAIN_LOGDIR}" \
  --dataset_dir="${PASCAL_DATASET}"
```

Along with this, I would comment out line 392 `flags.mark_flag_as_required('tf_initial_checkpoint')` in research/deeplab/train.py. 
As I said before, I also tried changing the learning rate, learning rate policy, optimization with Adam, but with all these experiments I did not manage to achieve results better than 0.4 miou.

**Comment by lillekemiker** [comment]
You probably want a larger batch size than 4. Increasing to 16 has helped me, albeit on a different dataset.

**Comment by asgill** [comment]
> You probably want a larger batch size than 4. Increasing to 16 has helped me, albeit on a different dataset.

Thanks Lillekemiker, I did not try using batch size of 16, but I did try batch size of 12, which did not help. My GPU is not able to support batch size greater than 12. I'll try to find a way around it.

I would appreciate any other suggestions!

**Comment by lillekemiker** [comment]
If increasing to 12 didn't make any difference, then that is probably not the only problem. I would stick with 12 if that is your max while playing around with other settings. I have not trained from scratch, myself, so I don't have much more help to offer right now. It might help to train the backbone separately before using it in this model.

**Comment by eerobert** [comment]
hi, 
would like to know how many step you had tried and what is the final loss value?
i am doing the same thing and would like to see how many iteration it takes to converge without using pretrained weights

**Comment by eerobert** [comment]
hi,
would like to know how many step you had tried and what is the final loss value?
i am doing the same thing and would like to see how many iteration it takes to converge without using pretrained weights

**Comment by HankKung** [comment]
I'm having the exact same problem when training nas_hnasnet. Have anyone solved it?

**Comment by PrinceP** [comment]
Can you try using the imagenet trained models to speed up the convergence? 
I am using that for a custom task and it is learning in 10,000 iterations. 
The modified task is on the separate domain itself.

**Comment by DianaLi96** [comment]
> I'm having the exact same problem when training nas_hnasnet. Have anyone solved it?

Have you solve the problem training on nas_hnasnet? I also met the problem. Thank you.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
